content_sync/backends/github.py

```diff
@@ -0,0 +1,153 @@
+""" Github backend """
+import logging
+
+from django.conf import settings
+from github import GithubException
+from github.Commit import Commit
+from github.ContentFile import ContentFile
+from github.PullRequest import PullRequest
+from github.Repository import Repository
+
+from content_sync.apis.github import GithubApiWrapper
+from content_sync.backends.api import check_sync_state
+from content_sync.backends.base import BaseSyncBackend
+from content_sync.models import ContentSyncState
+from websites.models import WebsiteContent
+
+
+log = logging.getLogger()
+
+
+class GithubBackend(BaseSyncBackend):
+    """
+    Github backend
+    """
+
+    def __init__(self, website: WebsiteContent):
+        """ Initialize the Github API backend for a specific website"""
+        self.api = GithubApiWrapper(website)
+        self.website = website
```

Comment:
Should probably do this instead, in case something in `BaseSyncBackend.__init__` changes in the future.
```suggestion
        super().__init__(website)
```